### PR TITLE
Update material, remove mdl and refactor

### PR DESCRIPTION
**Pull request by qdouble**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "clean": "npm cache clean && npm run rimraf -- node_modules doc typings coverage dist .awcache && npm run clean:compile",
     "clean:dist": "npm run rimraf -- dist .awcache",
     "clean:compile": "npm run rimraf -- \"src/compiled\" ",
+    "clean:overlaycss": "npm run rimraf -- \"node_modules/@angular/material/core/overlay/overlay.css\" ",
```
Comment (DzmitryShylovich) [inline package.json]: why is this required?

**Comment by DzmitryShylovich** [comment]
Does it work in aot? I'm asking because of https://github.com/angular/material2/pull/1378

File: package.json (hunk 1 above)
Comment (qdouble) [inline package.json]: By default, node-sass will throw an error about not being able to tell if it should import overlay.css or overlay.scss when you try to compile the material themes unless you delete overlay.css or manually edit the _core.scss file.

**Comment by qdouble** [comment]
Yes, it works with AOT, that was preventing me from updating it beforehand, but the bug was fixed in 9-3

**Comment by qdouble** [comment]
@DzmitryShylovich I haven't used the dialog/snackbar though, so not sure if an issue comes up there, but not getting any error with the starter files or my project

**Comment by qdouble** [comment]
This is the overlay.css issue right here: https://github.com/angular/material2/issues/1348

**Comment by DzmitryShylovich** [comment]
Ok thx 👍
